Service Connector  Hub/func.py

```diff
@@ -1,49 +1,58 @@
 import io
-import oci
-import re
 import os
 import json
-import sys
-import requests
 import logging
-import time
-import gzip
-from fdk import response
 
-def process(body):
+import requests
+
+
+def process(body: dict) -> None:
+    data = body.get("data", {})
+    source = body.get("source")
+    time = body.get("time")
+    dd_source = "oracle_cloud"
+    service = "OCI Logs"
+
+    # Get json data, time, and source information
+    payload = {}
+    payload.update({"source": source})
+    payload.update({"time": time})
+    payload.update({"data": data})
+    payload.update({"ddsource": dd_source})
+    payload.update({"service": service})
+
+    # Datadog endpoint URL and token to call the REST interface.
+    # These are defined in the func.yaml file.
     try:
-        data = body.get("data", {}) 
-        source = body.get("source") 
-        time = body.get("time")
-        ddsource = "oracle_cloud"
-        service = "OCI Logs"
-
-        #get json data, time, and source information
-        payload = {}
-        payload.update({"source":source}) 
-        payload.update({"time": time}) 
-        payload.update({"data":data})
-        payload.update({"ddsource": ddsource}) 
-        payload.update({"service":service})
-
-        #Datadog endpoint URL and token to call the REST interface. These are defined in the func.yaml file. 
-        datadoghost = os.environ['DATADOG_HOST']
-        datadogtoken = os.environ['DATADOG_TOKEN']
-
-        #Invoke Datadog API with the payload. If the payload contains more than one log this will be ingested as once. 
-        headers = {'Content-type': 'application/json', 'DD-API-KEY': datadogtoken}
-        x = requests.post(datadoghost, data = json.dumps(payload), headers=headers) 
-        logging.getLogger().info(x.text)
+        dd_host = os.environ['DATADOG_HOST']
+        dd_token = os.environ['DATADOG_TOKEN']
+        dd_tags = os.environ.get('DATADOG_TAGS', '')
+    except KeyError:
+        err_msg = "Could not find environment variables, \
+                   please ensure DATADOG_HOST and DATADOG_TOKEN \
+                   are set as environment variables."
+        logging.getLogger().error(err_msg)
+    if dd_tags:
+        payload.update({'ddtags': dd_tags})
 
+    # Invoke Datadog API with the payload.
+    # If the payload contains more than one log
+    # this will be ingested at once.
+    try:
+        headers = {'Content-type': 'application/json', 'DD-API-KEY': dd_token}
+        x = requests.post(dd_host, data=json.dumps(payload), headers=headers)
+        logging.getLogger().info(x.text)
```

Comment:
should we use Debug level here?

Reply:
we default to warn so info should be fine here.